File: backends/vapor/server/Sources/App/Web/Requests.swift
import Vapor

/// 请求体 DTO：字段全部可选，缺失/为 null 都先收下来，由 Validation 统一裁决
/// （这样"缺字段"也能返回契约要求的 422 + details，而不是解码层 4xx）。

struct RegisterRequest: Content {
    var email: String?
    var password: String?
    var nickname: String?
    var avatarId: String?
}

struct LoginRequest: Content {
    var email: String?
    var password: String?
}

struct RefreshRequest: Content {
    var refreshToken: String?
}

struct LogoutRequest: Content {
    var refreshToken: String?
}

struct UpdateProfileRequest: Content {
    var nickname: String?
    var avatarId: String?
}

struct ChangePasswordRequest: Content {
    var currentPassword: String?
    var newPassword: String?
}

struct CreateCapsuleRequest: Content {
    var title: String?
    var content: String?
    var openAt: String?
    var inPlaza: Bool?
}

struct FavoriteRequest: Content {
    var capsuleId: String?
}

struct CapsuleSuggestionRequest: Content {
    var title: String?
    var locale: String?
}

File: backends/axum/src/web/requests.rs
use serde::Deserialize;

/// 请求体 DTO：字段全部可选，缺失/为 null 都先收下来，由 Validation 统一裁决
/// （这样"缺字段"也能返回契约要求的 422 + details，而不是解码层 4xx）。

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RegisterRequest {
    pub email: Option<String>,
    pub password: Option<String>,
    pub nickname: Option<String>,
    pub avatar_id: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LoginRequest {
    pub email: Option<String>,
    pub password: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RefreshRequest {
    pub refresh_token: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LogoutRequest {
    pub refresh_token: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateProfileRequest {
    pub nickname: Option<String>,
    pub avatar_id: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ChangePasswordRequest {
    pub current_password: Option<String>,
    pub new_password: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCapsuleRequest {
    pub title: Option<String>,
    pub content: Option<String>,
    pub open_at: Option<String>,
    pub in_plaza: Option<bool>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FavoriteRequest {
    pub capsule_id: Option<String>,
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CapsuleSuggestionRequest {
    pub title: Option<String>,
    #[allow(dead_code)]
    pub locale: Option<String>,
}
